src/main/java/run/halo/sticker/endpoint/StickerAttachmentEndpoint.java

```diff
@@ -0,0 +1,190 @@
+import run.halo.app.core.extension.endpoint.CustomEndpoint;
+import run.halo.app.core.extension.service.AttachmentService;
+import run.halo.app.extension.GroupVersion;
+import run.halo.app.extension.Metadata;
+import run.halo.app.extension.ReactiveExtensionClient;
+import run.halo.app.plugin.ReactiveSettingFetcher;
+import run.halo.sticker.infra.StickerSetting;
+import run.halo.sticker.model.Sticker;
+import run.halo.sticker.model.StickerGroup;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class StickerAttachmentEndpoint implements CustomEndpoint {
+
+    private static final String SELF_USER = "-";
+    private static final String STICKER_GROUP_NAME = "sticker-group";
+    private static final String DEFAULT_STICKER_ATTACHMENT_POLICY_NAME = "default-policy";
+    private static final DataSize MAX_FILE_SIZE = DataSize.ofMegabytes(2L);
+    private final ReactiveExtensionClient client;
+    private final AttachmentService attachmentService;
+    private final ReactiveSettingFetcher settingFetcher;
+
+    @Override
+    public RouterFunction<ServerResponse> endpoint() {
+        var tag = "StickerV1alpha1Console";
+        return route()
+            // a get request resp test success text
+            .GET("sticker/test", this::test)
+            .POST("sticker/{groupName}/upload", contentType(MediaType.MULTIPART_FORM_DATA),
+                this::uploadUserSticker)
+            .build();
+    }
+
+    @Override
+    public GroupVersion groupVersion() {
+        return CustomEndpoint.super.groupVersion();
+    }
+
+    private Mono<ServerResponse> test(ServerRequest request) {
+        log.info("plugin-sticker test success");
+        return ServerResponse.ok().bodyValue("Hello, Sticker dev!" + new Date());
+    }
+
+    private Mono<ServerResponse> uploadUserSticker(ServerRequest request) {
+        var groupName = request.pathVariable("groupName");
+        log.info("Uploading sticker for user");
+        return request.body(BodyExtractors.toMultipartData())
+            .map(StickerUploadRequest::new)
+            .flatMap(this::uploadSticker)
+            .flatMap(attachment -> getOrCreateStickerGroup(groupName)
+                .flatMap(stickerGroup -> saveSticker(attachment)
+                    .doOnNext(sticker -> sticker.getSpec().setGroupName(stickerGroup.getMetadata().getName()))
+                    .flatMap(client::update)
+                )
+            )
+            .retryWhen(Retry.backoff(3, Duration.ofMillis(100))
+                .filter(throwable -> throwable instanceof OptimisticLockingFailureException))
+            .flatMap(sticker -> ServerResponse.ok().bodyValue(sticker));
+    }
+
+    private Mono<Sticker> saveSticker(Attachment attachment) {
+        var sticker = new Sticker();
+        sticker.setMetadata(attachment.getMetadata());
+        Sticker.StickerSpec stickerSpec = new Sticker.StickerSpec();
+        stickerSpec.setName(attachment.getMetadata().getName());
```

Comment:
建议改个名，name 表达的意义不明确。

```suggestion
        stickerSpec.setName(attachment.getMetadata().getName());
```